core/src/operations/mod.rs

```diff
@@ -156,6 +157,15 @@ pub enum Operation {
     /// instruction.
     Clk = OPCODE_CLK,
 
+    /// Emits an event id (`u32` value) to the advice provider.
+    ///
+    /// We interpret the event id as follows:
+    /// - 16 most significant bits identify the event source,
+    /// - 16 least significant bits identify the actual event.
+    ///
+    /// Similar to Noop, does not change the state of user stack.
+    Emit(u32) = OPCODE_EMIT,
```

Comment:
I would probably clarify that the immediate value affects program MAST root computation.